modules/@apostrophecms/rich-text-widget/ui/apos/apps/AposRichTextPermalinkResolver.js

```diff
@@ -0,0 +1,28 @@
+// This code is for EDITORS, to allow them to follow the permalinks
+// even though they are not replaced with the actual page URLs in
+// the markup on the server side. We do not do that for editors
+// because otherwise the permalink would be lost on the next edit.
+
+export default function() {
+  window.addEventListener('hashchange', e => {
+    // Typical case: hash link followed after page load
+    if (followPermalink()) {
+      e.stopPropagation();
+    }
+  });
+  // Or, the URL could already contain a permalink
+  followPermalink();
```

Comment:
Inside an A3 editor app file function, we can assume DOM ready has already occurred.